src/scheduler.rs

```diff
@@ -616,7 +616,7 @@ async fn fetch_idle(
             .await
             .context("delete_expired_imap_messages")?;
     } else if folder_config == Config::ConfiguredInboxFolder {
-        ctx.last_full_folder_scan.lock().await.take();
+        session.last_full_folder_scan.lock().await.take();
```

Comment:
Question: This is unrelated to this PR here, but still, I'm wondering:

What does this code do? As far as I understand, it does this:
- If we delete to the trash folder, but we don't know what the trash folder is
  - Then, _instead_ of fetching from the Inbox folder, we trigger a new full folder scan

@iequidoo or @link2xt do you know why?

Reply:
Yes, we must find Trash before fetching from Inbox, otherwise `fetch_move_delete()` will fail because Trash isn't configured. This is only needed for Gmail. Otherwise Trash will be never configured (we actually had such a bug before).